src/foam/nanos/crunch/connection/CapabilityPayloadDAO.js

```diff
@@ -120,24 +150,7 @@ foam.CLASS({
       
         // iterate through each capability from the select and calculate separate grant path
         for ( Capability rootCapability : rootCapabilities ){
-          List grantPath = crunchService.getGrantPath(x, rootCapability.getId());
-          Map<String,FObject> capabilityDataObjects = new HashMap<>();
-
-          // Adding separate capabilityDataObject map
-          for ( Object item : grantPath ) {
-            if ( item instanceof Capability ) {
-              Capability cap = (Capability) item;
-              FObject capDataObject = null;
-              if ( cap.getOf() != null ){
-                try {
-                  capDataObject = (FObject) cap.getOf().newInstance();
-                } catch (Exception e){
-                  throw new RuntimeException(e);
-                }
-              }
-              capabilityDataObjects.put(cap.getName(), capDataObject);
-            }
-          }
+          var capabilityDataObjects = walkGrantPath(crunchService.getGrantPath(x, rootCapability.getId()), x);
```

Comment:
either user var twice or use Map<> twice .. i.e. here and the call above in find_